src/features/game/events/landExpansion/bannerPurchased.ts

```diff
@@ -0,0 +1,68 @@
+import Decimal from "decimal.js-light";
+import cloneDeep from "lodash.clonedeep";
+import { GameState } from "../../types/game";
+import {
+  SEASONS,
+  SeasonalBanner,
+  getSeasonByBanner,
+  getSeasonalBanner,
+} from "features/game/types/seasons";
+
+export type PurchaseBannerAction = {
+  type: "banner.purchased";
+  name: SeasonalBanner;
+};
+
+type Options = {
+  state: Readonly<GameState>;
+  action: PurchaseBannerAction;
+  createdAt?: number;
+};
+
+export function purchaseBanner({
+  state,
+  action,
+  createdAt = Date.now(),
+}: Options): GameState {
+  const stateCopy = cloneDeep(state);
+  const { bumpkin, inventory } = stateCopy;
+
+  if (bumpkin === undefined) {
+    throw new Error("You do not have a Bumpkin");
+  }
+
+  if (inventory[action.name] !== undefined) {
+    throw new Error("You already have this banner");
+  }
+
```

Comment:
Consider also asserting `action.name` - we want to ensure it is a valid banner and not some other item in the game.

E.g. `if (!(name in BANNERS))`